### ISSUE TITLE
Launching server using command "racket" failed.

### ISSUE THREAD

**Issue by johnthompson-ybor** [Eugleo/magic-racket#131]
## Environment
- OS: MacOs
- OS Version [e.g. Catalina]
- VS Code Version [e.g. 1.46.1]
- Racket version [e.g. 7.7]
  - run: `racket -v`
  
  Welcome to Racket v8.15 [cs].
  
- Racket Langserver checksum [e.g. 64c1827548089c97c56a53a90ef3d569deb0e6c2]
  - run: `raco pkg show --full-checksum racket-langserver`
  
  raco pkg show --full-checksum racket-langserver
Installation-wide:
 [none]
User-specific for installation "8.15":
 Package            Checksum                                  Source
 racket-langserver  7481d2f98f175a545a0714ead3ec20ae327afec4  catalog...erver.git
 v0.6.7
 
- Magic Racket version [e.g. 0.5.6] 

## Error message
[Error - 4:20:59 PM] Starting client failed
Launching server using command "racket" failed.

I've tried hardcoding the path, it doesn't make any difference.  I can run racket and langserver and racket --lib racket-langserver just fine from the terminal.  There's no more useful debug information beyond that in anything I can find.

**Comment by jryans** [comment]
It seems like you skipped a few of the environment questions above...? Please fill them in so we have the full context on your situation.

**Comment by ilyapoz** [comment]
Same for me.

## Environment
- OS: MacOS
- OS Version Sonoma 14.4.1
- VS Code  1.95.3
- Racket version v 8.15 from homebrew.
- Racket Langserver checksum `7481d2f98f175a545a0714ead3ec20ae327afec4`
- Magic Racket version v0.6.7

## Error message
```
[Error - 20:47:22] Starting client failed
Launching server using command racket failed.
```

## Additional context
First intstallation. DrRacket works fine, tried to move to vscode, found magic racket, everything executes in the terminal, but lsp does not work.

**Comment by ilyapoz** [comment]
devconsole
```
notificationsAlerts.ts:42 Couldn't start client Racket Language Client
c @ notificationsAlerts.ts:42
log.ts:439   ERR spawn racket ENOENT: Error: spawn racket ENOENT
	at ChildProcess._handle.onexit (node:internal/child_process:286:19)
	at onErrorNT (node:internal/child_process:484:16)
	at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
console.ts:137 [Extension Host] rejected promise not handled within 1 second: Launching server using command racket failed.
uGi @ console.ts:137
log.ts:439   ERR Launching server using command racket failed.
```

**Comment by jryans** [comment]
Hmm, so far I am not able to reproduce this issue.

I am testing on a similar system configuration:

* macOS 14.7.1
* VS Code 1.95.3 
* Racket 8.15
* Racket Langserver `7481d2f98f175a545a0714ead3ec20ae327afec4`
* Magic Racket 0.6.7

These errors strongly suggest that the `racket` command is not accessible to VS Code somehow, but you've both said it works fine in the terminal...

Have you tried restarting VS Code just for good measure...? (Perhaps it's caching old binary paths...)

**Comment by ilyapoz** [comment]
Restarting did not help me. In my case racket is installed via homebrew if that matters `brew install --cask racket`

**Comment by ilyapoz** [comment]
Is there some diagnostic command we can try? In devtools or otherwise?

**Comment by jryans** [comment]
I also used Homebrew to install Racket via the cask.

One possible difference is that I am using an Intel Mac, while you may have Apple Silicon... though so far, I can't think of why that would matter here.

It's hard to know quite what to debug... The extension retrieves the LSP command and arguments from settings:

https://github.com/Eugleo/magic-racket/blob/8cf9d9a3195da7fd8078e019569ecaeec64608d9/src/utils.ts#L32-L37

and then passes them to the LSP client, which tries to run the command:

https://github.com/Eugleo/magic-racket/blob/8cf9d9a3195da7fd8078e019569ecaeec64608d9/src/extension.ts#L65-L70

The actual process launch is handled by the LSP client code.

What do you have `magicRacket.languageServer.command` and `magicRacket.languageServer.arguments` set to...? If you put those values together in a terminal and try to run it, that should be effectively the same as what the extension is doing.

**Comment by ilyapoz** [comment]
tl;dr

```
"magicRacket.languageServer.command": "/opt/homebrew/bin/racket",
```

fixes the problem in my case. I think previously I tried setting `magicRacket.general.racketPath` only and that did not (obviously) help.

To debug I created a python script on my machine and pointed the setting above to it and somehow homebrew bin is not in the `PATH`.

I guess homebrew sets PATH in some rc file that is not being sourced in the vscode plugins or whatnot. So most certainly not an issue with magicRacket.

I was able to format my files! Yay! 🎉 Thanks for the plugin, btw, using it to read SICP, much more fun than any other env.

**Comment by jryans** [comment]
Ah okay, glad you got it working! @johnthompson-ybor, maybe your issue is similar?

I will look into a better error message for this case, or perhaps displaying these values in context.
